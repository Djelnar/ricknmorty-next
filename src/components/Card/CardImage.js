import styled, { css } from 'styled-components'


export const CardImage = styled.img`
  display: block;
  width: 100%;
  object-fit: cover;

  ${(p) => p.link && css`
    cursor: pointer;
    transition-property: transform;
    transition-timing-function: ease-in;
    transition-duration: 150ms;
    &:hover {
      transform: scale(1.1);
      transition-timing-function: ease-out;
    }
  `}
`
