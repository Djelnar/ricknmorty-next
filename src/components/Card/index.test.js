/* global describe, it, expect */
import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'

import { Card } from './index'
import { CardImage } from './CardImage'


describe('Card Component', () => {
  it('should render correctly', () => {
    const output = renderer.create((
      <Card>
        <CardImage src="https://rickandmortyapi.com/api/character/avatar/1.jpeg" />
        <p>Rick</p>
        <p>and</p>
        <p>Morty</p>
      </Card>
    )).toJSON()

    expect(output).toMatchSnapshot()
  })
})
