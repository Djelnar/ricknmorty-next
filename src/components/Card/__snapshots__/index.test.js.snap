// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card Component should render correctly 1`] = `
.c0 {
  width: 100%;
  min-width: 150px;
  padding: 8px;
  height: 100%;
}

.c1 {
  background-color: #ffffff;
  padding: 8px;
  height: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 p {
  margin: 6px 0;
}

.c1 p:first-child {
  margin-top: 0;
}

.c1 p:last-of-type {
  margin-bottom: 14px;
}

.c2 {
  display: block;
  width: 100%;
  object-fit: cover;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      className="c2"
      src="https://rickandmortyapi.com/api/character/avatar/1.jpeg"
    />
    <p>
      Rick
    </p>
    <p>
      and
    </p>
    <p>
      Morty
    </p>
  </div>
</div>
`;

exports[`Card Component with Link prop should render correctly 1`] = `
.c0 {
  width: 100%;
  min-width: 150px;
  padding: 8px;
  height: 100%;
}

.c1 {
  background-color: #ffffff;
  padding: 8px;
  height: inherit;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
}

.c1 p {
  margin: 6px 0;
}

.c1 p:first-child {
  margin-top: 0;
}

.c1 p:last-of-type {
  margin-bottom: 14px;
}

.c2 {
  display: block;
  width: 100%;
  object-fit: cover;
  cursor: pointer;
  -webkit-transition-property: -webkit-transform;
  -webkit-transition-property: transform;
  transition-property: transform;
  -webkit-transition-timing-function: ease-in;
  transition-timing-function: ease-in;
  -webkit-transition-duration: 150ms;
  transition-duration: 150ms;
}

.c2:hover {
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  -webkit-transition-timing-function: ease-out;
  transition-timing-function: ease-out;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <img
      className="c2"
    />
  </div>
</div>
`;
